Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+        for (LatLong coord : pathCoords) {
+            pathPoints.add(CoordToLatLang(coord));
+        }
+
+        if (pathPoints.size() < 2) { // BaiduMap Polyline overlay need at least 2 points
+            if (mDroneLeashPath != null) {
+                mDroneLeashPath.remove();
+                mDroneLeashPath = null;
+            }
+            return;
+        }
+
+        if (mDroneLeashPath == null) {
+            PolylineOptions flightPath = new PolylineOptions();
+            flightPath.color(DRONE_LEASH_DEFAULT_COLOR).width(
+                    DroneHelper.scaleDpToPixels(DRONE_LEASH_DEFAULT_WIDTH,
+                            getResources()));
+            mDroneLeashPath = (Polyline)getBaiduMap().addOverlay(flightPath);
+        } else {
+            mDroneLeashPath.setPoints(pathPoints);
+        }
+    }
+
+    @Override
+    public void updateMissionPath(PathSource pathSource) {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+		List<LatLong> pathCoords = pathSource.getPathPoints();
+        final List<LatLng> pathPoints = new ArrayList<LatLng>(pathCoords.size());
+        for (LatLong coord : pathCoords) {
+            pathPoints.add(CoordToLatLang(coord));
+        }
+
+        if (pathPoints.size() < 2) { // BaiduMap Polyline overlay need at least 2 points
+            if (mMissionPath != null) {
+                mMissionPath.remove();
+                mMissionPath = null;
+            }
+            return;
+        }
+
+        if (mMissionPath == null) {
+            final PolylineOptions pathOptions = new PolylineOptions();
+            pathOptions.color(MISSION_PATH_DEFAULT_COLOR).width(MISSION_PATH_DEFAULT_WIDTH).points(pathPoints);
+            mMissionPath = (Polyline) map.addOverlay(pathOptions);
+        } else {
+            mMissionPath.setPoints(pathPoints);
+        }
+    }
+
+
+    @Override
+    public void updatePolygonsPaths(List<List<LatLong>> paths) {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+        for (Polygon poly : polygonsPaths) {
+            poly.remove();
+        }
+
+        for (List<LatLong> contour : paths) {
+            PolygonOptions pathOptions = new PolygonOptions();
+            pathOptions.fillColor(POLYGONS_PATH_DEFAULT_COLOR);
+			final List<LatLng> pathPoints = new ArrayList<LatLng>(contour.size());
+            for (LatLong coord : contour) {
+                pathPoints.add(CoordToLatLang(coord));
+            }
+            pathOptions.points(pathPoints);
+            polygonsPaths.add((Polygon)map.addOverlay(pathOptions));
+        }
+
+    }
+
+    @Override
+    public void addCameraFootprint(FootPrint footprintToBeDraw) {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+        PolygonOptions pathOptions = new PolygonOptions();
+		pathOptions.stroke(new Stroke(FOOTPRINT_DEFAULT_WIDTH, FOOTPRINT_DEFAULT_COLOR));
```

Comment:
@droneboost store `new Stroke(...)` into a constant since it's unlikely to change.